src/mds/journal.cc

```diff
@@ -520,6 +521,9 @@ void EMetaBlob::fullbit::decode(bufferlist::const_iterator &bl) {
   if (struct_v >= 9) {
     decode(alternate_name, bl);
   }
+  if (struct_v >= 10) {
+    decode(fscrypt_last_block, bl);
```

Comment:
I'm not sure if this is the place to do it, but we probably do want to add a mds_fscrypt_last_block_max_size tunable or something so that the client can't send us truncate requests with an enormous fscrypt_last_block. We could probably make the default 4k, but since it's a max, you could make it larger (8k or so).

Reply:
Okay, will add that.